R/test.data.table.R

```diff
@@ -1,27 +1,34 @@
-test.data.table <- function(verbose=FALSE, pkg="pkg", silent=FALSE, with.other.packages=FALSE, benchmark=FALSE, Rraw=NULL) {
-  if (exists("test.data.table",.GlobalEnv,inherits=FALSE)) {
+test.data.table <- function(verbose=FALSE, pkg="pkg", silent=FALSE, with.other.packages=FALSE, benchmark=FALSE, script=NULL) {
+  if (exists("test.data.table", .GlobalEnv,inherits=FALSE)) {
     # package developer
     # nocov start
     if ("package:data.table" %chin% search()) stop("data.table package is loaded. Unload or start a fresh R session.")
-    d = if (pkg %chin% dir()) file.path(getwd(), pkg) else Sys.getenv("CC_DIR")
-    d = file.path(d, "inst/tests")
+    rootdir = if (pkg %chin% dir()) file.path(getwd(), pkg) else Sys.getenv("CC_DIR")
+    subdir = file.path("inst","tests")
     # nocov end
   } else {
     # i) R CMD check and ii) user running test.data.table()
-    d = paste0(getNamespaceInfo("data.table","path"),"/tests")
+    rootdir = getNamespaceInfo("data.table","path")
+    subdir = "tests"
   }
-  # for (fn in dir(d,"*.[rR]$",full=TRUE)) {  # testthat runs those
+  fulldir = file.path(rootdir, subdir)
 
-  if (!is.null(Rraw)) {
-    stopifnot(is.character(Rraw), length(Rraw)==1L, !is.na(Rraw), nzchar(Rraw))
-    fn = Rraw
+  if (!is.null(script)) {
+    stopifnot(is.character(script), length(script)==1L, !is.na(script), nzchar(script))
+    if (!identical(basename(script), script)) {
+      subdir = dirname(script)
+      fulldir = normalizePath(subdir, mustWork=FALSE)
+      fn = basename(script)
+    } else {
+      fn = script
+    }
   } else {
     stopifnot( !(with.other.packages && benchmark) )
     fn = if (with.other.packages) "other.Rraw"
          else if (benchmark) "benchmark.Rraw"
          else "tests.Rraw"
   }
-  fn = file.path(d, fn)
+  fn = setNames(file.path(fulldir, fn), file.path(subdir, fn))
```

Comment:
`fulldir` is full path, while when printing summary we might want to retain current better looking `inst/test/file.Rraw` rather than full path. Thus this nice path is now stored as name of `fn`.